system-reinstall-bootc/src/podman.rs

```diff
@@ -58,3 +59,42 @@ pub(crate) fn command(image: &str, root_key: &Option<UserKeys>) -> Command {
 
     command
 }
+
+pub(crate) fn ensure_podman_installed() -> Result<()> {
+    if Command::new("podman").arg("--version").can_execute() {
+        return Ok(());
+    }
+
+    tracing::warn!("Podman was not found on this system. It's required to install a bootc image.");
+
+    ensure!(
+        Command::new("dnf").arg("--version").can_execute(),
+        "Please install podman and try again."
+    );
+
+    let mut dnf_install_command = Command::new("dnf");
+    dnf_install_command
+        .arg("install")
+        // TODO: Is this a bad idea? It answers yes to EVERYTHING. Maybe we should let the
+        // user interact with dnf manually instead... I wish dnf gave us more granular
+        // control over what we can automatically answer yes to.
+        .arg("-y")
+        .arg("podman");
+
+    if prompt::ask_yes_no(
+        format!("Would you like to install podman using dnf? ({dnf_install_command:?})").as_str(),
```

Comment:
I'm not opposed to asking, but on the other hand, why would the user doing this answer no?

Combining the above two things I think we could unconditionally just invoke `/usr/lib/system-reinstall-bootc/install-podman`.

Or maybe I guess we allow the `install-podman` script to be interactive, we pass it stdin, and don't use `-y` for dnf by default there.

But I think for our use case here we can just install by default.

Reply:
>why would the user doing this answer no?

IDK, seems polite to ask before installing a package